packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/dev_server/proxy.rb

```diff
@@ -101,6 +105,22 @@ def secure_session_id
 
         private
 
+        def clean_sfr_cache(env, query, headers)
+          return if @cache_cleaned
+
+          @cache_cleaned = true
+
+          query = query.dup
+          query << ["preview_theme_id", theme_id]
+
+          request(
+            env["REQUEST_METHOD"], env["PATH_INFO"],
```

Comment:
I'm not 100% sure how that works. Why would doing the request twice but once with the query param fix the problem? It seems like we're throwing away the result of this request too. Do we need this to clear a remote cache (and not a local one?)